pubsub/README.rst

```diff
@@ -41,6 +39,96 @@ Here's the rough usage pattern for using ``SubscriberClient``:
         max_messages=10)
 
 
+There's also ``gcloud.aio.pubsub.subscribe`` helper function you can use to
+setup a pubsub processing pipeline. It is built with ``asyncio`` and thus only
+available in ``gcloud-aio-pubsub`` package. The usage is fairly simple:
+
+.. code-block:: python
+
+    from gcloud.aio.pubsub import SubscriberClient
+    from gcloud.aio.pubsub import subscribe
+    from gcloud.aio.pubsub.metrics_agent import MetricsAgent
+
+    subscriber_client = SubscriberClient()
+
+    async def handler(message):
+        return
+
+    await subscribe(
+        'projects/<my_project>/subscriptions/<my_subscription>',
+        handler,
+        subscriber_client,
+        num_producers=1,
+        max_messages_per_producer=100,
+        ack_window=0.3,
+        num_tasks_per_consumer=1,
+        metrics_client=MetricsAgent()
+    )
+
+While defaults are somewhat sensible, it is highly recommended to performance
+test your application and tweak function parameter to your specific needs.
+Here's a few hints:
+
+:``handler``:
+    an async function that will be called for each message. It should accept an
+    instance of ``SubscriberMessage`` as its only argument and return ``None``
+    if the message should be acked. An exception raised within the handler will
```

Comment:
A thought..

Is it worth calling out how one should go about explicitly `nack`ing a message? i.e. via modify_ack_deadline = 0. Take `egress` for example. `egress` has a pretty lengthy ack deadline of 20 seconds because its downstream service can have some long running requests. On top of this, `egress` also processes messages very quickly (aside from the time it takes to POST them), and also retries a lot of messages. I suspect we may want to explicitly nack messages in the case of failures in `egress`' case because of this to reduce potential latency as a result of retrying.

This may have been mentioned in the pr already but there are so many comments now 😄

Reply:
I think explicit nacking just naturally creeped out of scope for this one. That said, now when I'm testing egress it looks like we really need this in version 1.

Reply:
Created this as a separate PR to not clutter this one. Please do take a look! https://github.com/talkiq/gcloud-aio/pull/270